testing/src/main/java/org/oppia/android/testing/math/PolynomialSubject.kt

```diff
@@ -0,0 +1,156 @@
+package org.oppia.android.testing.math
+
+import com.google.common.truth.FailureMetadata
+import com.google.common.truth.IntegerSubject
+import com.google.common.truth.StringSubject
+import com.google.common.truth.Truth.assertAbout
+import com.google.common.truth.Truth.assertThat
+import com.google.common.truth.Truth.assertWithMessage
+import com.google.common.truth.extensions.proto.LiteProtoSubject
+import org.oppia.android.app.model.Polynomial
+import org.oppia.android.testing.math.PolynomialSubject.Companion.assertThat
+import org.oppia.android.testing.math.RealSubject.Companion.assertThat
+import org.oppia.android.util.math.getConstant
+import org.oppia.android.util.math.isConstant
+import org.oppia.android.util.math.toPlainText
+
+// TODO(#4100): Add tests for this class.
+
+/**
+ * Truth subject for verifying properties of [Polynomial]s.
+ *
+ * Note that this class is also a [LiteProtoSubject] so other aspects of the underlying [Polynomial]
+ * proto can be verified through inherited methods.
+ *
+ * Call [assertThat] to create the subject.
+ */
+class PolynomialSubject(
+  metadata: FailureMetadata,
+  private val actual: Polynomial?
+) : LiteProtoSubject(metadata, actual) {
+  private val nonNullActual by lazy {
+    checkNotNull(actual) {
+      "Expected polynomial to be defined, not null (is the expression/equation not a valid" +
+        " polynomial?)"
+    }
+  }
+
+  /** Verifies that the represented [Polynomial] is null (i.e. not a valid polynomial). */
+  fun isNotValidPolynomial() {
+    assertWithMessage(
+      "Expected polynomial to be undefined, but was: ${actual?.toPlainText()}"
+    ).that(actual).isNull()
+  }
+
+  /**
+   * Verifies that the represented [Polynomial] is a constant (i.e. [Polynomial.isConstant] and
+   * returns a [RealSubject] to verify the value of the constant polynomial.
+   */
+  fun isConstantThat(): RealSubject {
```

Comment:
The prefix we are using `*That` sounds a bit different to me. I don't have any specific suggestions for this.

Reply:
sounds good.